metrics/lib/features/dashboard/data/model/build_data.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:metrics/features/dashboard/data/model/data_model.dart';
+import 'package:metrics/features/dashboard/domain/entities/core/build.dart';
+import 'package:metrics/features/dashboard/domain/entities/core/percent.dart';
+
+/// [DataModel] that represents the [Build] entity.
+class BuildData extends Build implements DataModel {
+  const BuildData({
+    String id,
+    DateTime startedAt,
+    BuildStatus result,
+    Duration duration,
+    String workflowName,
+    String url,
+    Percent coverage,
+  }) : super(
+          id: id,
+          startedAt: startedAt,
+          result: result,
+          duration: duration,
+          workflowName: workflowName,
+          url: url,
+          coverage: coverage,
+        );
+
+  /// Creates the [BuildData] from the [json] and it's [id].
+  factory BuildData.fromJson(Map<String, dynamic> json, String id) {
+    final buildResultValue = json['result'] as int;
+    final durationMilliseconds = json['duration'] as int;
+
+    return BuildData(
+      id: id,
+      startedAt: (json['startedAt'] as Timestamp).toDate(),
+      result: BuildStatus.values[buildResultValue ?? 0],
```

Comment:
Should we store as text rather than relying on the index? Also maybe it should be null or unknown instead of 0 (which is "successful"?)?